src/typeahead/index.tsx

```diff
@@ -170,17 +187,27 @@ export const Typeahead = factory(function Typeahead({
 
 				const allItems = get({ query: getOptions().query });
 				if (allItems && allItems.length >= activeIndex) {
-					const { itemDisabled = (item: ListOption) => item.disabled } = properties();
+					const { itemDisabled } = properties();
 
 					const activeItem = allItems[activeIndex];
+					let disabled = false;
+					if (activeItem) {
+						disabled = itemDisabled ? itemDisabled(activeItem) : !!activeItem.disabled;
 
-					if (!itemDisabled(activeItem)) {
-						const { onValue } = properties();
-
-						icache.set('value', activeItem.value);
-						onClose();
-						onValue(activeItem.value);
-						icache.set('lastValue', activeItem.value);
+						if (!disabled) {
+							icache.set('value', activeItem.value);
+							onClose();
+							callOnValue(activeItem.value);
+						}
+					} else {
+						if (strict) {
+							icache.set('valid', false);
+							const { onValidate } = properties();
+							onValidate && onValidate(false);
```

Comment:
I sort of feel like if we're in strict mode and a user hits "enter" and no values are found, we should just no-op rather than failing validation.

Reply:
I've added an `if` so that this only happens if it's required